<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StreetGenerator</class>
 <widget class="QMainWindow" name="StreetGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1384</width>
    <height>988</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>StreetGenerator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>650</y>
      <width>941</width>
      <height>271</height>
     </rect>
    </property>
    <property name="title">
     <string>Inputs</string>
    </property>
    <widget class="GLImagePreview" name="hMapRender">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>300</width>
       <height>200</height>
      </rect>
     </property>
    </widget>
    <widget class="GLImagePreview" name="popMapRender">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>20</y>
       <width>300</width>
       <height>200</height>
      </rect>
     </property>
    </widget>
    <widget class="GLImagePreview" name="geogMapRender">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>20</y>
       <width>300</width>
       <height>200</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="labelHMap">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Height Map: ......</string>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonHMap">
     <property name="geometry">
      <rect>
       <x>244</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Browse</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelPMap">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>230</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Population Density...</string>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonPMap">
     <property name="geometry">
      <rect>
       <x>554</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Browse</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelGMap">
     <property name="geometry">
      <rect>
       <x>640</x>
       <y>230</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Geography: ...</string>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonGMap">
     <property name="geometry">
      <rect>
       <x>864</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Browse</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdClearHeight">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdClearPop">
     <property name="geometry">
      <rect>
       <x>490</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdClearGeog">
     <property name="geometry">
      <rect>
       <x>800</x>
       <y>230</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>251</width>
      <height>641</height>
     </rect>
    </property>
    <property name="title">
     <string>Inputs</string>
    </property>
    <widget class="QComboBox" name="comboBox">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <item>
      <property name="text">
       <string>Grid</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Street Style</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>90</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Various Inputs....</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>650</y>
      <width>411</width>
      <height>271</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
      <horstretch>1</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="title">
     <string>Controls</string>
    </property>
    <widget class="QPushButton" name="cmdGenerateStreets">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>190</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdSaveOutput">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>230</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Save...</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Various Output Parameters</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdSaveOutputImage">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>230</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Save Image</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>161</width>
       <height>161</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Render Options</string>
        </property>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>14</y>
           <width>141</width>
           <height>121</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QRadioButton" name="radShowPopulation">
            <property name="text">
             <string>Show Population</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radShowGeography">
            <property name="text">
             <string>Show Geography</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radShowHeightmap">
            <property name="text">
             <string>Show Heightmap</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radShowNone">
            <property name="text">
             <string>Show None</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="chkShowVerts">
            <property name="text">
             <string>Show Vertices</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="cmdReset">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>110</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Initialise / Reset</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cmdStep">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>150</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Step...</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>236</x>
       <y>40</y>
       <width>171</width>
       <height>70</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="2" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Seed:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="intSeedValue">
        <property name="minimum">
         <number>-1000000000</number>
        </property>
        <property name="maximum">
         <number>1000000000</number>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="chkTimeSeed">
        <property name="text">
         <string>Use system time</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QStreetRenderWidget" name="streetRender">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>20</y>
      <width>1101</width>
      <height>631</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
      <horstretch>1</horstretch>
      <verstretch>1</verstretch>
     </sizepolicy>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1384</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuWindows">
    <property name="title">
     <string>Windows</string>
    </property>
    <addaction name="actionBuilding_Generator"/>
    <addaction name="actionTexture_Generator"/>
   </widget>
   <addaction name="menuWindows"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionBuilding_Generator">
   <property name="text">
    <string>Building Generator</string>
   </property>
  </action>
  <action name="actionTexture_Generator">
   <property name="text">
    <string>Texture Generator</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLImagePreview</class>
   <extends>QOpenGLWidget</extends>
   <header>glimagepreview.h</header>
  </customwidget>
  <customwidget>
   <class>QStreetRenderWidget</class>
   <extends>QFrame</extends>
   <header>qstreetrenderwidget.h</header>
   <container>1</container>
   <slots>
    <slot>setDrawVertices(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonPMap</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_buttonPMap_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>600</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>663</x>
     <y>963</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonGMap</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_buttonGMap_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>907</x>
     <y>923</y>
    </hint>
    <hint type="destinationlabel">
     <x>953</x>
     <y>967</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdGenerateStreets</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>onClickGenerate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1275</x>
     <y>892</y>
    </hint>
    <hint type="destinationlabel">
     <x>1134</x>
     <y>956</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radShowPopulation</sender>
   <signal>toggled(bool)</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_radPopClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1023</x>
     <y>764</y>
    </hint>
    <hint type="destinationlabel">
     <x>955</x>
     <y>707</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radShowGeography</sender>
   <signal>toggled(bool)</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_radGeogClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1060</x>
     <y>790</y>
    </hint>
    <hint type="destinationlabel">
     <x>956</x>
     <y>738</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radShowHeightmap</sender>
   <signal>toggled(bool)</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_radHeightClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1086</x>
     <y>813</y>
    </hint>
    <hint type="destinationlabel">
     <x>954</x>
     <y>772</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkShowVerts</sender>
   <signal>toggled(bool)</signal>
   <receiver>streetRender</receiver>
   <slot>setDrawVertices(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1018</x>
     <y>859</y>
    </hint>
    <hint type="destinationlabel">
     <x>966</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdReset</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>onClickReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1321</x>
     <y>811</y>
    </hint>
    <hint type="destinationlabel">
     <x>1378</x>
     <y>797</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdStep</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>onClickStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1339</x>
     <y>848</y>
    </hint>
    <hint type="destinationlabel">
     <x>1374</x>
     <y>832</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdSaveOutput</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>onClickSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1339</x>
     <y>918</y>
    </hint>
    <hint type="destinationlabel">
     <x>1377</x>
     <y>921</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdSaveOutputImage</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>onClickSaveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1162</x>
     <y>924</y>
    </hint>
    <hint type="destinationlabel">
     <x>1164</x>
     <y>964</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radShowNone</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>on_radNoneClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1014</x>
     <y>839</y>
    </hint>
    <hint type="destinationlabel">
     <x>955</x>
     <y>809</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdClearHeight</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>clearHeightmap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>923</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>956</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdClearPop</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>clearPopmap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>924</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>955</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdClearGeog</sender>
   <signal>clicked()</signal>
   <receiver>StreetGenerator</receiver>
   <slot>clearGeogmap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>844</x>
     <y>921</y>
    </hint>
    <hint type="destinationlabel">
     <x>843</x>
     <y>959</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_buttonPMap_clicked()</slot>
  <slot>on_buttonGMap_clicked()</slot>
  <slot>onClickGenerate()</slot>
  <slot>on_radPopClicked()</slot>
  <slot>on_radGeogClicked()</slot>
  <slot>on_radHeightClicked()</slot>
  <slot>onClickReset()</slot>
  <slot>onClickStep()</slot>
  <slot>onClickSaveImage()</slot>
  <slot>onClickSave()</slot>
  <slot>on_radNoneClicked()</slot>
  <slot>clearHeightmap()</slot>
  <slot>clearGeogmap()</slot>
  <slot>clearPopmap()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
